design/specs/table_inspector.md

```diff
@@ -0,0 +1,205 @@
+## Enabling and Disabling the Table Inspector
+
+The table inspector can be enabled or disabled manually or automatically based on event triggers.
+
+To manually enable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to show the table inspector.
+
+Some events will automatically display the table inspector:
+
+- Navigating to a newly created table after completing the data import process.
+- Navigating to a newly created table after completing the column extraction process.
+
+To disable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to hide the table inspector.
+
+### Positioning
+
+When users enable the table inspector, a panel will appear on the right side of the screen, sharing the same container as the table and pushing its contents rather than being stacked on top of it.
+
+In some cases, this will cause the table portion of the layout to have a horizontal scroll.
+
+### Inspector Modes
+
+At the top of the inspector panel is a row of tabs. Each tab includes table component-specific attributes and actions.
+
+The tabs are listed in the following order:
+
+- Table (Default)
+- Column
+- Record
+- Cell
+
+## Using the Table Inspector
+
+Selecting different inspector modes in the table inspector allows users to examine at the properties, options and settings of table components. Aside from that, various actions can also be initiated through the inspector.
+
+### Table Mode
+
+Shown when inspector mode is set to `Table`.
+
+Properties:
+
+- Table Name
+- Table Descriptions
+
+Settings:
+
+- Record Summary
+- Table Links (Constraints)
+
+Actions:
+
+- Remove Duplicates
+- Create Query from Table
+  
+![image](/assets/design/specs/table_inspector/183378847-b942ecd3-6f3c-4cd4-8cc4-3080a041b2a4.png)
+
+![image](/assets/design/specs/table_inspector/183380108-b3db8d3b-7301-4a19-bc38-cd777a4dde46.png)
+
+### Column Mode
+
+Shown when inspector mode is set to `Column`.
+
+Properties:
+
+- Column Name
+- If Column is Link:
+  - Link Source
+
+Options:
+
+- Allow NULL
+- Allow Duplicates
+- Data Type
+  - Database Options
+  - Format Options
+
+Actions:
+
+- Extract Columns to Table
+
+![image](/assets/design/specs/table_inspector/183432613-ead6315e-3802-4345-9427-820c66094797.png)
+  
+![image](/assets/design/specs/table_inspector/183433425-9a906a91-28a4-4045-9300-732af811ed8d.png)
+
+### Record Mode
+
+Shown when inspector mode is set to `Record`.
+
+Actions:
+
+- Duplicate Record
+
+![image](/assets/design/specs/table_inspector/183444464-fb268bfc-77e2-45cf-9180-373cf950ca63.png)
+
+### Cell Mode
+
+Shown when inspector mode is set to `Cell`.
+
+![image](/assets/design/specs/table_inspector/183445209-a2d7bf2c-453b-4cae-84e8-94b645ce9271.png)
+
+### Multi-Select Mode
+
+Shown when multiple table components are selected. The table inspector will display the properties according to the current inspector mode and the shared properties between selected components.
```

Comment:
This is unclear, I'm not sure what it means.

Reply:
@kgodey This no longer applies. It was relevant when the inspector states were dependent on current selection. I will remove it.